src/Type.cpp

```diff
@@ -931,10 +984,11 @@ Type::make_all_struct_types_with_bitfields(Enumerator<string> &enumerator,
 
 	int bitfields_cnt = 0;
 	int normal_fields_cnt = 0;
+	bool hasAssignOps = if_struct_will_have_assign_ops();
```

Comment:
The definition of ```hasAssignOps``` is far from where it's used. Can we move this statement to above the following line?

    bool hasImplicitNontrivialAssignOps = hasAssignOps || ...